### ISSUE TITLE
false positive in wps361 , inconsistently structured comprehension

### ISSUE THREAD

**Issue by Sxderp** [wemake-services/wemake-python-styleguide#2202]
### What's wrong

I'm not sure why this would be invalid.

```python
[
        os.path.realpath(os.path.expanduser(conffile))
        for conffile in (
            os.environ.get('HOMESYSADM_CONF_REPO', ''),
            os.environ.get('HOMESYSADM_CONF_SITE', ''),
            *reversed(os.environ.get('HOMESYSADM_CONF_USER', '').split(':')),
        )
        if conffile
]
```

### How it should be

I'd like my comprehension to work?

### Flake8 version and plugins

{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.6.8",
    "system": "Linux"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-bandit",
      "version": "2.1.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-broken-line",
      "version": "0.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-bugbear",
      "version": "21.9.2"
    },
    {
      "is_local": false,
      "plugin": "flake8-comprehensions",
      "version": "3.6.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-darglint",
      "version": "1.8.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-debugger",
      "version": "4.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-docstrings",
      "version": "1.6.0, pydocstyle: 6.1.1"
    },
    {
      "is_local": false,
      "plugin": "flake8-eradicate",
      "version": "1.1.0"
    },
    {
      "is_local": false,
      "plugin": "flake8-string-format",
      "version": "0.3.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_commas",
      "version": "2.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_isort",
      "version": "4.0.0"
    },
    {
      "is_local": false,
      "plugin": "flake8_quotes",
      "version": "3.3.0"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "naming",
      "version": "0.11.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.7.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.3.1"
    },
    {
      "is_local": false,
      "plugin": "pylint",
      "version": "0.0.0"
    },
    {
      "is_local": false,
      "plugin": "rst-docstrings",
      "version": "0.2.3"
    },
    {
      "is_local": false,
      "plugin": "wemake_python_styleguide",
      "version": "0.15.3"
    }
  ],
  "version": "3.9.2"
}

### pip information

appdirs==1.4.4
astor==0.8.1
attrs==21.2.0
bandit==1.7.0
chardet==3.0.4
colorama==0.4.4
coverage==5.5
darglint==1.8.0
decorator==4.0.11
distlib==0.3.1
distro==1.2.0
dnspython==1.16.0
docutils==0.17.1
entrypoints==0.3
eradicate==2.0.0
filelock==3.0.12
flake8==3.9.2
flake8-bandit==2.1.2
flake8-broken-line==0.3.0
flake8-bugbear==21.9.2
flake8-commas==2.0.0
flake8-comprehensions==3.6.1
flake8-debugger==4.0.0
flake8-docstrings==1.6.0
flake8-eradicate==1.1.0
flake8-isort==4.0.0
flake8-polyfill==1.0.2
flake8-quotes==3.3.0
flake8-rst-docstrings==0.2.3
flake8-string-format==0.3.0
flakehell==0.9.0
gitdb==4.0.7
GitPython==3.1.18
gssapi==1.3.0
gtarcsbluecat==1.4.0
gtbluecatsync==1.3.4
gtldap==3.0.0b0
gtsnipeapi==2.0.0a1
idna==2.10
importlib-metadata==3.1.1
importlib-resources==3.3.0
isort==5.9.3
Jinja2==2.11.1
ldap3==2.8.1
libnacl==1.7.2
M2Crypto==0.35.2
MarkupSafe==0.23
mccabe==0.6.1
msgpack==0.6.2
netaddr==0.7.18
nose==1.3.7
nosexcover==1.0.11
pbr==5.6.0
pep8-naming==0.11.1
psutil==5.6.7
pyasn1==0.4.7
pycodestyle==2.7.0
pycrypto==2.6.1
pycurl==7.43.0
pydocstyle==6.1.1
pyflakes==2.3.1
Pygments==2.10.0
pygobject==3.22.0
pyroute2==0.4.13
PySocks==1.6.8
PyYAML==5.4.1
pyzmq==17.0.0
requests==2.14.2
restructuredtext-lint==1.3.2
rfc3986==1.3.0
rpm==4.11.3
salt==3002.6
six==1.14.0
smmap==4.0.0
snowballstemmer==2.1.0
stevedore==3.4.0
testfixtures==6.18.3
toml==0.10.2
typing-extensions==3.10.0.2
urllib3==1.25.6
virtualenv==20.2.2
wemake-python-styleguide==0.15.3
zipp==3.4.0

### OS information

RHEL 7.9

**Comment by sobolevn** [comment]
Thanks for the report!

**Comment by sobolevn** [comment]
`WPS361` is removed.
